dsl-reference.md

```diff
@@ -28,6 +28,7 @@
     - [Set](#set)
     - [Try](#try)
     - [Wait](#wait)
+    - [Debug](#debug)
```

Comment:
I'm honestly wondering if this is a `task` or a `call`.

Reply:
Typically you will call debug within a composite task before and after invoking another task. so I think is easier if debug is a task itself.

Reply:
It would probably be even better to call it via an extension.

Reply:
I agree with Jb, imho it should not be a task nor a call.

Reasons are:
1. It's irrelevant to authors of the workflow (but might be to implementers): there is no effectivement on the workflow
2. There is no way to unify behavior accross runtimes (logs will have different format and features. Runtimes might enforce structured logs, too)
3. You can achieve the same using shell+echo or, even better, outputting logs to a specific file
4. Runtime specific: aside from 2, runtimes (ex: Synapse) might not want to offer the user the ability to write to logs. If It's up to the runtimes to get logs from Somewhere and do smtg with it, it does not belong to the DSL imo.

Reply:
Furthermore, the use case you describe to @ricardozanini is covered by the sample logging extension (see dsl-reference.md)

Reply:
Bottom line is that I would instead recommend that "log" task to become the first one in the functions gallery. So instead of "forcing" runtimes to implement it, we create a 'run' task which uses shell to echo or output a potentially formatred message.
Wdyt?

Reply:
See https://github.com/serverlessworkflow/specification/blob/main/dsl.md#creating-a-custom-function

Reply:
@JBBianchi I want to avoid different runtimes implementing logging in different ways. 
In your extension example, you are invoking an URI to collect logging, but this task is intended to be able to call the runtime library  (a logging library in Java, fmt package in go, Ilogger in C#) that actually logs in the console without losing portability (if any runtime implemenation support logging, the definition file is fully portable)

Reply:
@cdavernas 
Sample logging extension is an example of how to collect logs, it does not cover the case where you want to log a message to the standard console. I think using extension mechanism is superb if we have the log task, but without the log task, how do users logs to the console?
I think forcing people to write logs though a call to a shell (which actually assume the workflow is running with access to a shell, which is not true if we are using windows SO) is worse that forcing runtime to implement a log task. 
As a side note, as I mentioned in the issue, this is not my personal preference, I actually have real feedback from users ot the current spec.  They are using custom sysout function (sonataflow) a lot. This makes workflows not portable. Im trying to avoid losing portability and losing functionality (I cannot tell that users that they cannot log to the console any longer)

Reply:
> 2\. There is no way to unify behavior accross runtimes (logs will have different format and features. Runtimes might enforce structured logs, too)

Any programming language which is used to implement the spec will have means to write to standar output (if there is not standard output, nothing is written and thats all), so implementation is trivial, 
The message is just the string, the format is up to the implementation (which is fine as far as the message is part of the log). Implementations might provide extra configuration to format the log message (up to the implementation) The point is that the workflow definition will still be portable and when executed in any implementation it will be provide some kind of feedback though the standard console (if available). Thats all.